src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -1205,6 +1210,9 @@ private Task DestroyActivations(List<ActivationData> list)
                         directory.InvalidateCacheEntry(activationData.Address);
 
                         RerouteAllQueuedMessages(activationData, null, "Finished Destroy Activation");
+
+                        activationData.ServiceScope?.Dispose();
+                        activationData.ServiceScope = null;
```

Comment:
Would it be better to make `IActivationData` an `IDisposable` and keep this logic there?